zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -449,13 +449,23 @@ where
     ) -> Self {
         // Prevent loss of miner funds due to an unsupported or incorrect address type.
         if let Some(miner_address) = mining_config.miner_address.clone() {
-            assert_eq!(
-                miner_address.network(),
-                network.clone(),
-                "incorrect miner address config: {miner_address} \
-                         network.network {network} and miner address network {} must match",
-                miner_address.network(),
-            );
+            match network.kind() {
+                network_kind @ NetworkKind::Mainnet => assert_eq!(
+                    miner_address.network_kind(),
+                    NetworkKind::Mainnet,
+                    "Incorrect mining address config: Zebra is configured to run on a {network_kind} network, \
+                    but the configured mining address is for {}. The network kinds must match.",
```

Comment:
We have a bug in the message for non-Mainnet network kinds, so I tried to adjust this one so that they look similar.

```suggestion
                    "Incorrect config: Zebra is configured to run on a Mainnet network, \
                    which implies the configured mining address needs to be for Mainnet, \
                    but the provided address is for {}.",
```

Reply:
Applied in https://github.com/ZcashFoundation/zebra/pull/8383/commits/fb2d13a7062d5d65db49cf89a472425c9efd59bd.